src/client/auth-client.ts

```diff
@@ -496,6 +501,62 @@ export class AuthClient {
     checkInterruption();
   }
 
+  public async loginWithIcrc29(options?: AuthClientLoginOptions): Promise<void> {
+    // Merge the passed options with the options set during creation
+    const loginOptions = mergeLoginOptions(this._createOptions?.loginOptions, options);
+
+    // Set default maxTimeToLive to 8 hours
+    const maxTimeToLive = loginOptions?.maxTimeToLive ?? DEFAULT_MAX_TIME_TO_LIVE;
+
+    // Create the URL of the IDP. (e.g. https://XXXX/authorize?openid=<selected-open-id>)
+    const identityProviderUrl = new URL(
+      loginOptions?.identityProvider?.toString() || IDENTITY_PROVIDER_DEFAULT,
+    );
+    // Set the correct pathname
+    identityProviderUrl.pathname = IDENTITY_PROVIDER_ICRC29_PATH;
```

Comment:
Current IDENTITY_PROVIDER_DEFAULT doesn't support ICRC-29 and we probably shouldn't override the path name, a custom supplied identity provider might have a different pathname than `/authorize`.

```suggestion
    // Create the URL of the IDP. (e.g. https://XXXX/authorize?openid=<selected-open-id>)
    const identityProviderUrl = new URL(
      loginOptions?.identityProvider?.toString() || IDENTITY_PROVIDER_ICRC29_DEFAULT,
    );
```

Reply:
Good point